config/initializers/devise.rb

```diff
@@ -27,3 +27,34 @@
 
   config.timeout_in = 7.days
 end
+
+Warden::Manager.after_set_user do |record, warden, options|
+  next unless FeatureFlag.active?(:incident_eviction)
+
+  scope = options[:scope]
+  lra = warden.session(scope)['last_request_at']
+
+  # User is not affected
+  next unless record.id.in?(1..46) && record.instance_of?(Candidate)
+
+  # The cookie must have a last_request_at in order to be relevant
+  next if lra.nil?
+
+  case lra
+  when Integer
+    last_request_at = Time.zone.at(lra)
+  when String
+    last_request_at = Time.zone.parse(lra)
```

Comment:
Just curious, can this be an integer **or** a string? Or is it always one or the other?

Reply:
Yeah, it's a bit odd, but this is how Devise has to handle this as well. I think it must depend on the browser / session somehow.